src/tclmisc.c

```diff
@@ -248,48 +248,72 @@ static int tcl_binds STDVAR
 
 static int tcl_timer STDVAR
 {
-  unsigned long x;
-  char s[26];
+  char *x;
 
-  BADARGS(3, 4, " minutes command ?count?");
+  BADARGS(3, 5, " minutes command ?count ?name??");
 
   if (atoi(argv[1]) < 0) {
     Tcl_AppendResult(irp, "time value must be positive", NULL);
     return TCL_ERROR;
   }
-  if (argc == 4 && atoi(argv[3]) < 0) {
+  if (argc >= 4 && atoi(argv[3]) < 0) {
     Tcl_AppendResult(irp, "count value must be >= 0", NULL);
     return TCL_ERROR;
   }
   if (argv[2][0] != '#') {
-    x = add_timer(&timer, atoi(argv[1]), (argc == 4 ? atoi(argv[3]) : 1),
-                  argv[2], 0L);
-    snprintf(s, sizeof s, "timer%lu", x);
-    Tcl_AppendResult(irp, s, NULL);
+    if (argc == 5) {
+      /* Prevent collisions with unnamed timers */
+      if (strncmp(argv[4], "timer", 5) == 0) {
+        Tcl_AppendResult(irp, "timer name may not begin with \"timer\"", NULL);
+        return TCL_ERROR;
+      }
+      /* Check for existing timers by same name */
+      if (find_timer(timer, argv[4])) {
+        Tcl_AppendResult(irp, "timer already exists by that name", NULL);
+        return TCL_ERROR;
+      }
+    }
+    x = add_timer(&timer, atoi(argv[1]), (argc >= 4 ? atoi(argv[3]) : 1),
+                  argv[2], (argc == 5 ? argv[4] : '\0'), 0L);
```

Comment:
argv[4] is char* but '\0' is char, which should be an error. you can use "" for the empty string or pass NULL if that works